File: JavaScript Fundamentals/Mid Exams/Mid Exam 6/treasureHunt.js
function solve(arr) {
    let loot = arr.shift().split('|');
    for (let el of arr) {
        if (el == 'Yohoho!') break;

        let tokens = el.split(' ');
        if (tokens[0] == 'Loot') {
            for (let index = 1; index < tokens.length; index++) {
                if (!(loot.includes(tokens[index]))) loot.splice(0, 0, tokens[index]);
            }
        } else if (tokens[0] == 'Drop') {
            let index = Number(tokens[1]);
            if (index >= 0 && index < loot.length) {
                let removedItem = loot.splice(index, 1);
                loot.push(removedItem);
            }
        } else if (tokens[0] == 'Steal') {
            let count = Number(tokens[1]);
            let stolenItems = [];
            for (let i = 1; i <= count; i++) {
                if (loot.length > 0) stolenItems.push(loot.pop());
            }
            console.log(stolenItems.reverse().join(', '));
        }
    }
    if (loot.length == 0) console.log('Failed treasure hunt.');
    else {
        let totalLength = 0;
        for (let item of loot) {
            totalLength += item.length;
        }
        console.log(`Average treasure gain: ${(totalLength / loot.length).toFixed(2)} pirate credits.`);
    }
}

File: Python Fundamentals/Mid Exam problems/treasure_hunt.py
initial_loot = input().split('|')

while True:
    command = input()
    if command == "Yohoho!":
        break
    command_list = command.split()
    if command_list[0] == 'Loot':
        for loot_item in command_list[1:]:
            if loot_item not in initial_loot:
                initial_loot.insert(0, loot_item)

    elif command_list[0] == 'Drop':
        if 0 <= int(command_list[1]) <= len(initial_loot) - 1:
            initial_loot.append(initial_loot.pop(int(command_list[1])))

    elif command_list[0] == 'Steal':
        stolen_items = []
        for stolen_item in range(int(command_list[1])):
            stolen_items.append(initial_loot.pop(-1))
            if len(initial_loot) == 0:
                break
        stolen_items.reverse()
        print(', '.join(stolen_items))

if initial_loot:
    items_total_lengths = sum([len(x) for x in initial_loot])
    print(f"Average treasure gain: {items_total_lengths / len(initial_loot):.2f} pirate credits.")
else:
    print("Failed treasure hunt.")
